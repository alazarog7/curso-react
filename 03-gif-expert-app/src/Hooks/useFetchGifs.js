import { useEffect, useState } from "react";
import { getGifs } from "../Helpers/GetGifs";

export const useFetchGifs = ( category ) => {
    const [state, setstate] = useState({
        data:[],
        loading: true
    });

    console.log(category);

    useEffect(()=>{
        getGifs(category).then((imgs)=> {
             setstate({
                 data: imgs,
                 loading: false
             });
         }).catch(()=>{
            setstate({
                data: [],
                loading: true
            });
         })
     },[category]);

    return state;

};