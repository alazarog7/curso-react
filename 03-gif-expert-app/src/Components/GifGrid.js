import React, { useState, useEffect } from 'react';
import { useFetchGifs } from '../Hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';
import { PropTypes } from 'prop-types';

const GifGrid = ({category}) => {

    //const [images, setImages] = useState([]);
    const { data:images, loading} = useFetchGifs(category);
   
    return (
        <>
            <h3 className="animate__zoomInLeft">{category}</h3>
            {loading && <p>Cargando ...</p> }
            <div className="card-grid">
                {
                    images.map( img => <GifGridItem key={img.id} {...img}/> )
                }
            </div>
        </>
    );
};

export default GifGrid;


GifGrid.propTypes={
    category: PropTypes.string.isRequired
}

