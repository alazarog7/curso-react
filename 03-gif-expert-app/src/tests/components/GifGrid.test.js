import React from 'react';
import shallow from "enzyme/build/shallow"
import GifGrid from "../../Components/GifGrid"
import { useFetchGifs } from '../../Hooks/useFetchGifs';

jest.mock('../../Hooks/useFetchGifs')

describe('<GifGrid />', () => {
    
    const category = 'homero';

    test('should return a snapshot', () => {
        useFetchGifs.mockReturnValue({
            data:[],
            loading: true
        });
        const wrapper = shallow(<GifGrid category={category}/>);
        expect(wrapper).toMatchSnapshot();
    })

    test('should show items when the images load', () => {
        
        const gifs = [{
            id: 'ABC',
            url: 'www.google.com',
            title: 'cualquier cosa'
        },
        {
            id: '123',
            url: 'www.yahoo.com',
            title: 'cualquier cosa 2'
        }
        ];
        useFetchGifs.mockReturnValue({
            data:gifs,
            loading: false
        });

        const wrapper = shallow(<GifGrid category={category}/>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);     
    })
    
    
})
