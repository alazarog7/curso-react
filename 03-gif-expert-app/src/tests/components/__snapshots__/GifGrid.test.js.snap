// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifGrid /> should return a snapshot 1`] = `
<Fragment>
  <h3
    className="animate__zoomInLeft"
  >
    homero
  </h3>
  <p>
    Cargando ...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`<GifGrid /> should show items when the images load 1`] = `
<Fragment>
  <h3
    className="animate__zoomInLeft"
  >
    homero
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="cualquier cosa"
      url="www.google.com"
    />
    <GifGridItem
      id="123"
      key="123"
      title="cualquier cosa 2"
      url="www.yahoo.com"
    />
  </div>
</Fragment>
`;
